torchvision/prototype/features/_image.py

```diff
@@ -71,36 +77,33 @@ def __new__(
         device: Optional[Union[torch.device, str, int]] = None,
         requires_grad: bool = False,
     ) -> Image:
-        data = torch.as_tensor(data, dtype=dtype, device=device)
-        if data.ndim < 2:
+        tensor = cls._to_tensor(data, dtype=dtype, device=device, requires_grad=requires_grad)
+        if tensor.ndim < 2:
             raise ValueError
-        elif data.ndim == 2:
-            data = data.unsqueeze(0)
-        image = super().__new__(cls, data, requires_grad=requires_grad)
+        elif tensor.ndim == 2:
+            tensor = tensor.unsqueeze(0)
 
         if color_space is None:
-            color_space = ColorSpace.from_tensor_shape(image.shape)  # type: ignore[arg-type]
+            color_space = ColorSpace.from_tensor_shape(tensor.shape)  # type: ignore[arg-type]
             if color_space == ColorSpace.OTHER:
                 warnings.warn("Unable to guess a specific color space. Consider passing it explicitly.")
         elif isinstance(color_space, str):
             color_space = ColorSpace.from_str(color_space.upper())
         elif not isinstance(color_space, ColorSpace):
             raise ValueError
-        image.color_space = color_space
 
-        return image
-
-    def __repr__(self, *, tensor_contents: Any = None) -> str:  # type: ignore[override]
-        return self._make_repr(color_space=self.color_space)
+        return cls._wrap(tensor, color_space=color_space)
 
     @classmethod
-    def new_like(
-        cls, other: Image, data: Any, *, color_space: Optional[Union[ColorSpace, str]] = None, **kwargs: Any
-    ) -> Image:
-        return super().new_like(
-            other, data, color_space=color_space if color_space is not None else other.color_space, **kwargs
+    def wrap_like(cls, other: Image, tensor: torch.Tensor, *, color_space: Optional[ColorSpace] = None) -> Image:
```

Comment:
This signature seems strange. IMO, we should not override attributes defined by `other`. Let's remove color_space arg. Same for other `wrap_like` methods.

Reply:
> IMO, we should not override attributes defined by `other`

That is the whole point of this method. This should not only wrap the tensor, but also provide a convenient interface to change the metadata. If we wouldn't do that, why would we have the `other` reference in the first place? This is on par with the `torch.*_like` methods that let you override the `dtype` or `device` during creation:

```pycon
>>> t = torch.rand(3, dtype=torch.float64)
>>> torch.rand_like(t, dtype=torch.float16)
tensor([0.4985, 0.7021, 0.7393], dtype=torch.float16)
```

What you seem to be proposing here is to have a "bare metal" function. That is what `_wrap` is currently doing. We could just make it public.

Reply:
I think changing the signature is worth it. I missed it on the original review and I must agree with Victor this looks a bit weird.

Reply:
> I think changing the signature is worth it.

Changing the signature to what? If we remove all the metadata overwrites how are we going to do something like 

https://github.com/pytorch/vision/blob/7eb5d7fcab73afec976907a855d9e63fa31f5579/torchvision/prototype/features/_bounding_box.py#L87

where we need to retain the `format` of the current bounding box, but need to change the `image_size`?

Reply:
As discuss offline, we should either remove it from everywhere or from none. We decided offline to keep as-is.